### DISCUSSION TITLE
Where did the CSV of all danbooru tags come from?

### DISCUSSION THREAD

**Discussion by PaperOrb** [DominikDoom/a1111-sd-webui-tagcomplete#194] [Q&A] [answered]
edit: I did find this, though I don't have nor want a google account, so I'll be sticking to the CSV that comes with the extension unless there's something else. https://danbooru.donmai.us/forum_topics/12774

I'm wanting a CSV copy for personal use. I've been using the copy that comes installed in the extensions folder with this extension, but it looks like it stops cleanly at 100k tags which makes me skeptical that some tags are missing (actually I know there's been tags on danbooru that weren't searchable in the extension).

**Comment by DominikDoom** [comment, marked as answer]
For a long time danbooru had a public daily dump under https://danbooru.donmai.us/cache/tags.json, and a similar one for aliases.
However, this was removed some time at the end of 2022, probably to discourage data scraping. But that is where my copy came from, I think my version is a snapshot from November. It was **all** tags on danbooru at that point as far as I know, including deprecated ones.

I then converted the json file to an SQLite database using https://github.com/dcmoura/spyql so I could query and export it easier and simply created a view with the top 100k tags sorted by post count and joined with their aliases, then exported that to csv.

You are correct that there are a few tags not included, but it should only be tags either added after the snap was taken or with a very low post count, since as said above deprecated tags are included (a few old tags have a lot of associated posts before the tag was changed).
I also have a full csv and a few with other cutoffs, but chose to not use those as they either included a lot of irrelevant tags with 0 to 20 posts or cut off too much. The full size is 1,071,088 rows in the tag table and 26,104 rows for the aliases, but a *lot* of those are tags with 0 posts. >0 is ~500k, >10 already only ~150k.

I can send you the full SQLite DB if you want, which is probably a lot easier to work with than the CSVs. It contains not everything that was in the original JSON, but most of it.
Here's a link to it: https://pastebin.com/XM6rRfb9 (will expire in one day, just so I don't get too much traffic on that file)

**Reply by PaperOrb** [reply]
I guess these sites that require user content and unpaid volunteers to function don't see the hypocrisy in gate keeping metadata of all things... Rant aside, much appreciated. I downloaded the db just in case, but I'll stick to the CSV for now since I've only ever dealt with sqlite for a student project a few years ago. My query requirements are rather modest anyway since I basically just need fuzzy search to filter sets of tags so I can export them to separate new CSV's that will then later get converted into txt files to be used as wildcards. Thanks again for all that you do

**Reply by DominikDoom** [reply]
If you just want to quickly view or export stuff from the DB without further editing, I can highly recommend [DB Browser for SQLite](https://sqlitebrowser.org), which is a small but great tool for both quick and in depth stuff. It helps to know SQL, but you can browse and filter quite a bit without having to write a query.
